### ISSUE TITLE
Support RMSE with Uncertainty

### ISSUE THREAD

**Issue by singh-kristian** [dreamquark-ai/tabnet#585]
## Feature request
Currently TabNet Regression does not offer an option to explicitly model the uncertainty in the regression outputs. Other regression implementations of popular algorithms (e.g. [catboost](https://catboost.ai/docs/en/concepts/loss-functions-regression#RMSEWithUncertainty)) support this uncertainty estimation.

**What is the expected behavior?**
An option for a new class TabNetUncertaintyRegressor, that will produce mean and variance regression estimates, and using [GNLL](https://docs.pytorch.org/docs/stable/generated/torch.nn.GaussianNLLLoss.html#gaussiannllloss) for a loss function.

**What is motivation or use case for adding/changing the behavior?**
For use-cases where model uncertainty is needed for decision making / actioning model results, or when predicting a gaussian distribution is the required output.

**How should this be implemented in your opinion?**
A new class should be created from the base tab model that has two output heads (one for mean and one for log var). GNLL can be used as the training loss, and RMSE can still be reported for a eval metric. Additionally information on variance prediction spread would be useful across training samples. #

**Are you willing to work on this yourself?**
yes (currently have a working version of an implementation)
